components/api/validator/validator.bal

```diff
@@ -0,0 +1,72 @@
+import ballerina/config;
+import ballerina/http;
+import ballerina/log;
+import ballerina/time;
+import ballerina/io;
+
+boolean isIntrospectionEPInitialized = false;
+
+http:Client introspectionEP = new("https://wso2.com" , config = {
+        auth: {
+            scheme: http:BASIC_AUTH,
+            config: {
+                username: "",
+                password: ""
+            }
+        }
+    });
+# Description
+#
+# + token - access token to be validated
+# + conf - authorization configuration struct
+# + return - returns whether the access token is valid or not
+public function getUsername(string token, Conf conf) returns (string)|error {
```

Comment:
In a case if token is invalid, I don't see any reason which is logged. Whether the username mismatch or token expiary.

Reply:
As discussed in the review I am only checking the token is active or not. I am not validating it with the username or expiary.

Reply:
AFAIK we just discussed about validating expiry time in the review. However validating username is a must.

Reply:
Is the username header is a mandatory one?